src/Core/Foundation/Version.php

```diff
@@ -391,4 +393,14 @@ private function removeLegacyPrefix($version, $majorVersionString)
 
         return $version;
     }
+
+    /**
+     * Checks if the version number matches old PS version styles (1.X.X)
+     */
+    private static function matchLegacyPattern(string $version): bool
+    {
+        $result = preg_match('/^1\.(.*)$/', $version); // version starting with "1."
```

Comment:
Seems to be false, `1.2` or `1.2.3` are not legacy pattern, the patter is four digits beginning with one :thinking:

Reply:
it's an understanding issue here

For me the old way to define versions (legacy) was wrong because we didn't increment the good digit on every major

after 1.0, the next major would be 2.0 and not 1.1

So I assume that anyversion number starting with `1.` is the legacy way of defining version

Reply:
I agree with @PierreRambaud : we should consider legacy version numbers need 4 digits

Reply:
I'm not sure the four numbers should be mandatory, for example `1.7.5` is a legacy version even if the patch part is missing

Reply:
Pierre is right